lib/Dialect/TritonGPU/Transforms/Pipeliner/MatmulLoopPipeline.cpp

```diff
@@ -518,21 +517,22 @@ assignMemoryLayouts(scf::ForOp &forOp,
     loadsToPipeline.insert(&op);
     LoadInfo loadInfo;
     for (auto use : users) {
+      // By default we will try pipelining with load to registers at the end.
+      // For mmav3 we can try leaving the operands in shared memory.
+      bool mmav3Shmem = false;
       if (isa<mlir::triton::DotOpInterface>(use)) {
         LDBG("set shared encoding with dot user: " << *use);
-        auto mmaLoadType = getMMALoadType(&op);
         auto dot = dyn_cast<tt::DotOp>(use);
         auto warpGroupDot = dyn_cast<ttng::WarpGroupDotOp>(use);
+        mmav3Shmem = canUseMMAv3Pipelining(&op) && warpGroupDot;
```

Comment:
I think this breaks for MMA v5. We are seeing numerical errors and bad IR after this PR